jr-stree/src/main/java/com/fasterxml/jackson/jr/stree/JacksonJrsTreeCodec.java

```diff
@@ -50,6 +65,16 @@ private JrsValue nodeFrom(JsonParser p) throws IOException
             return JrsBoolean.FALSE;
         case JsonTokenId.ID_NUMBER_INT:
         case JsonTokenId.ID_NUMBER_FLOAT:
+            final JsonParser.NumberType n = p.getNumberType();
+            if (n == JsonParser.NumberType.FLOAT) {
+                return new JrsNumber(p.getFloatValue());
+            }
+            if (_useBigDecimalForFloat && n == JsonParser.NumberType.DOUBLE) {
```

Comment:
second check unnecessary; if flag is set, `FLOAT` would also be coerced not just `DOUBLE`.

Reply:
done!